pkg/odo/cli/component/push.go

```diff
@@ -76,12 +75,16 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 	po.localConfig = conf
 	po.sourceType = conf.LocalConfig.GetSourceType()
 
+	glog.V(0).Infof("SourceLocation: %s", po.localConfig.GetSourceLocation())
+
 	// Get SourceLocation here...
-	po.sourcePath, err = conf.GetOSSourcePath()
+	po.sourcePath, err = conf.GetOSSourcePath(po.componentContext)
```

Comment:
`po.componentContext` can be a relative path, and by default, it is (`./`) so before using it for `GetOsSourcePath` will have to convert it to the full abolute path. `util.GetAbsPath` might be handy for that.